lib/tasks_private/regions.rake

```diff
@@ -0,0 +1,35 @@
+# frozen_string_literal: true
+
+namespace 'aws:extract' do
+  desc 'Get / renew regions list from online source of AWS gem'
+  task :regions do
+    require 'uri'
+    require 'json'
+    require 'yaml'
+    require 'net/http'
+
+    # path valid for aws-sdk v3 gem's source
+    uri = URI('https://raw.githubusercontent.com/aws/aws-sdk-ruby/master/gems/aws-partitions/partitions.json')
+    out_file = ManageIQ::Providers::Amazon::Engine.root / 'lib' / 'data' / 'aws_regions.yaml'
+    service_name = 'ec2'
```

Comment:
ec2-only feels like it may be insufficient, because we also connect to s2, sqs, sns, etc, but it might be ok if ec2 is always a superset of the others.

Reply:
It was the same before, please take a look at the previous module code version. All those URLs are about `ec2`.